UnityProject/Assets/Scripts/Health/RespiratoryHealth/RespiratorySystem.cs

```diff
@@ -268,6 +370,177 @@ private void HandleBreathingPlasma(float plasmaAmount)
 				string.Format(plasmaHighOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
 			);
 		}
+		return plasmaConsumed;
+	}
+
+	private bool HandleBreathingCarbonDioxide(GasMix gasMix)
+	{
+		float carbonPressure = gasMix.GetPressure(Gas.CarbonDioxide);
+		// there is a little carbon dioxide in the air.
+		if (carbonPressure <= CARBON_DIOXIDE_SAFE_MAX && carbonPressure > CARBON_DIOXIDE_WARNING_LEVEL)
+		{
+			if (DMMath.Prob(90))
+			{
+				return true;
+			}
+
+			// 10% chances of message
+			var theirPronoun = gameObject.Player() != null
+				? gameObject.Player().Script.characterSettings.TheirPronoun()
+				: "its";
+			Chat.AddActionMsgToChat(
+				gameObject,
+				CO2LowYouMessages.PickRandom(),
+				string.Format(
+					CO2LowOthersMessages.PickRandom(),
+					gameObject.ExpensiveName(),
+					string.Format(plasmaLowOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun))
+			);
+		}
+		// enough carbon dioxide to start suffocating you!
+		else if (carbonPressure > CARBON_DIOXIDE_SAFE_MAX)
+		{
+			if (CarbonSuffocationTimer >= 0)
+				CarbonSuffocationTimer++;
+			if (CarbonSuffocationTimer >= 3 && CarbonSuffocationTimer < 7)
+			{
+				IsSuffocating = true;
+				bloodSystem.OxygenDamage += 3;
+
+				if (DMMath.Prob(90))
+				{
+					return true;
+				}
+
+				// 10% chances of message
+				var theirPronoun = gameObject.Player() != null
+					? gameObject.Player().Script.characterSettings.TheirPronoun()
+					: "its";
+				Chat.AddActionMsgToChat(
+					gameObject,
+					CO2MedYouMessages.PickRandom(),
+					string.Format(CO2MedOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
+				);
+			}
+			else if (CarbonSuffocationTimer >= 7)
+			{
+				IsSuffocating = true;
+				bloodSystem.OxygenDamage += 8;
+
+				if (DMMath.Prob(90))
+				{
+					return true;
+				}
+
+				// 10% chances of message
+				var theirPronoun = gameObject.Player() != null
+					? gameObject.Player().Script.characterSettings.TheirPronoun()
+					: "its";
+				Chat.AddActionMsgToChat(
+					gameObject,
+					CO2HighYouMessages.PickRandom(),
+					string.Format(CO2HighOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
+				);
+				return true;
+			}
+		}
+
+		if (carbonPressure < CARBON_DIOXIDE_WARNING_LEVEL)
+		{
+			CarbonSuffocationTimer = 0;
+			return (carbonPressure > 0);
+		}
+		return false;
+	}
+	private bool HandleWearingGasMask(GasMix gasMix)
+	{
+
+		// if there is too much CO2 in the air
+		if (gasMix.GetMoles(Gas.CarbonDioxide) >= 30)
+		{
+			// if there is too much CO2 in the air, but there isn't too much plasma in the air
+			if (gasMix.GetMoles(Gas.Plasma) < 25)
+			{
+				GasMix gasMix2 = gasMix;
+				gasMix2.RemoveGas(Gas.CarbonDioxide, 30);
+				HandleBreathingCarbonDioxide(gasMix2);
+				return true;
+			}
+			else if (gasMix.GetMoles(Gas.Plasma) >= 25)
```

Comment:
Isn't the check here redundant?

Reply:
good point, i'll remove the redundancies.